Source/WebCore/platform/xr/PlatformXR.h

```diff
@@ -289,18 +289,38 @@ class Device : public ThreadSafeRefCounted<Device>, public CanMakeWeakPtr<Device
             template<class Decoder> static std::optional<StageParameters> decode(Decoder&);
         };
 
+#if USE(MTLTEXTURE_FOR_XR_LAYER_DATA)
+        struct SharedTexture {
+            mutable MachSendRight handle = { };
```

Comment:
The mutable here is working around your original problem.
If the call to startFrame passes the ownership of the object and thus its subobjects, then it is more appropriate to just pass the data as rvalue reference, so that the ownership transfer is communicated that way.

Reply:
Moving the data in via `startFrame` is a little complicated.